src/main/java/com/megansportfolio/budgettracker/report/ReportService.java

```diff
@@ -0,0 +1,165 @@
+import java.util.stream.Collectors;
+
+@Service
+public class ReportService {
+
+    @Autowired
+    BudgetItemUpdateDao budgetItemUpdateDao;
+
+    @Autowired
+    BudgetDao budgetDao;
+
+    @Autowired
+    BudgetEntryDao budgetEntryDao;
+
+    //set all values not in the controller
+    public List<BudgetItem> getReportBudgetItems(long budgetId, int year, String loggedInUserEmailAddress){
+        Optional<Budget> budget = budgetDao.findById(budgetId);
+        if(budget.isPresent()){
+            if(!budget.get().getUser().getUsername().equals(loggedInUserEmailAddress)){
+                throw new RuntimeException();
+            }
+            else {
+                List<BudgetItem> budgetItems = budget.get().getBudgetItems();
+                for (BudgetItem budgetItem : budgetItems) {
+                    setYearlyBudgetedAmount(budgetItem, year);
+                    setYearlyAmountSpent(budgetItem, year);
+                    setPercentDifference(budgetItem);
+                    setAverageMonthlyDifference(budgetItem);
+                }
+                return budgetItems;
+            }
+        }
+        else{
+            return null;
+        }
+    }
+
+    //get with updates yearly budgeted, BI
+    public void setYearlyBudgetedAmount(BudgetItem budgetItem, int year){
+        BigDecimal yearlyBudgetedAmount = BigDecimal.ZERO;
+        List<BudgetItemUpdate> budgetItemUpdates = budgetItemUpdateDao.findAllByBudgetItemIdAndYear(budgetItem.getId(), year);
+
+        if(budgetItem.getBudgetItemType() == BudgetItemType.ANNUAL){
+            if(budgetItemUpdates.isEmpty()){
+                yearlyBudgetedAmount = budgetItem.getAmount();
+            }
+            else{
+                List<BudgetItemUpdate> sortedUpdates = budgetItemUpdates.stream()
+                        .sorted(Comparator.comparing(BudgetItemUpdate::getDate).reversed())
+                        .collect(Collectors.toList());
+                yearlyBudgetedAmount = sortedUpdates.get(0).getAmount();
+            }
+        }
+        else {
+            for(int i = 1; i <= 12; i++){
+                Optional<BudgetItemUpdate> correspondingUpdate = BudgetItemUpdateService.findCorrespondingBudgetItemUpdate(year, i, budgetItemUpdates);
+                if(correspondingUpdate.isPresent()){
+                    yearlyBudgetedAmount = yearlyBudgetedAmount.add(correspondingUpdate.get().getAmount());
+                }
+                else{
+                    yearlyBudgetedAmount = yearlyBudgetedAmount.add(budgetItem.getAmount());
+                }
+            }
+        }
+        budgetItem.setYearlyBudgetedAmount(yearlyBudgetedAmount);
+    }
+
+    //get yearly spent, BI
+    public void setYearlyAmountSpent(BudgetItem budgetItem, int year){
+        BigDecimal yearlyAmountSpent = BigDecimal.ZERO;
+        List<BudgetEntry> budgetEntries = budgetEntryDao.findAllByBudgetItemIdAndYear(budgetItem.getId(), year);
+        if(budgetEntries.isEmpty()){
```

Comment:
You can simplify lines 91-99 a bit:
`if(!budgetEntries.isEmpty()){  // calculate yearly amount spent }`
`budgetItem.setYearlyAmountSpent(yearlyAmountSpent);`